extensions/data-transfer/portability-data-transfer-daybook/src/main/java/org/datatransferproject/transfer/daybook/social/DaybookPostsImporter.java

```diff
@@ -0,0 +1,164 @@
+    Map<String, String> linkMap = new HashMap<>();
+
+    String content = activity.getContent() == null ? "" : activity.getContent();
+    String title = activity.getTitle() == null ? "" : activity.getTitle();
+    String location =
+        activity.getLocation() == null || activity.getLocation().getName() == null
+            ? ""
+            : activity.getLocation().getName();
+    String published =
+        activity.getPublished().toString() == null ? "" : activity.getPublished().toString();
+
+    Request.Builder requestBuilder = new Request.Builder().url(baseUrl);
+    requestBuilder.header("token", authData.getAccessToken());
+
+    FormBody.Builder builder = new FormBody.Builder().add("type", "POSTS");
+    builder.add("exporter", JobMetadata.getExportService());
+    builder.add("content", content);
+    builder.add("title", title);
+    builder.add("location", location);
+    builder.add("published", published);
+
+    Collection<SocialActivityAttachment> linkAttachments =
+        activity.getAttachments().stream()
+            .filter(attachment -> attachment.getType() == SocialActivityAttachmentType.LINK)
+            .collect(Collectors.toList());
+    Collection<SocialActivityAttachment> imageAttachments =
+        activity.getAttachments().stream()
+            .filter(attachment -> attachment.getType() == SocialActivityAttachmentType.IMAGE)
+            .collect(Collectors.toList());
+
+    // Just put link attachments at the bottom of the post, as we
+    // don't know how they were laid out in the originating service.
+    if (!linkAttachments.isEmpty()) {
+
+      for (SocialActivityAttachment attachment : linkAttachments) {
+        linkMap.put(attachment.getName(), attachment.getUrl());
+      }
+      try {
+        String json = objectMapper.writeValueAsString(linkMap);
+        builder.add("link", json);
+      } catch (JsonProcessingException e) {
+        monitor.info(() -> String.format("Error processing JSON: %s", e.getMessage()));
+      }
+    }
+
+    if (!imageAttachments.isEmpty()) {
+
+      for (SocialActivityAttachment image : imageAttachments) {
+        imageMap.put(image.getName() != null ? image.getName() : image.getUrl(), image.getUrl());
+      }
+      try {
+        String json = objectMapper.writeValueAsString(imageMap);
+        builder.add("image", json);
+      } catch (JsonProcessingException e) {
+        monitor.info(() -> String.format("Error processing JSON: %s", e.getMessage()));
+      }
+    }
+
+    FormBody formBody = builder.build();
+    requestBuilder.post(formBody);
+
+    Response response = client.newCall(requestBuilder.build()).execute();
+    int code = response.code();
+    // Though sometimes it returns error code for success requests
+    if (code < 200 || code > 299) {
+      throw new IOException(
+          String.format(
+              "Error occurred in request for adding entry, message: %s", response.message()));
+    }
+
+    return response.message();
```

Comment:
It would be good to enclose this response object in a try-with-resources block:

```suggestion
    try (Response response = client.newCall(requestBuilder.build()).execute()) {
      int code = response.code();
      // Though sometimes it returns error code for success requests
      if (code < 200 || code > 299) {
        throw new IOException(
            String.format(
                "Error occurred in request for adding entry, message: %s", response.message()));
      }

      return response.message();
    }
```